code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm

```diff
@@ -185,18 +185,18 @@
 	animate(src, pixel_x = 0, pixel_z = 0, time = 10)
 	sleep(10)
 	playsound(src.loc, 'sound/effects/meteorimpact.ogg', 200, 1)
-	for(var/mob/living/L in range(1,tturf))
-		if(L == src)
-			continue
+	for(var/mob/living/L in orange(1, src))
 		if(L.stat)
 			visible_message("<span class='danger'>[src] slams down on [L], crushing them!</span>")
 			L.gib()
 		else
-			var/throwtarget = get_edge_target_turf(src, get_dir(src, get_step_away(L, src)))
 			L.adjustBruteLoss(75)
-			L.throw_at_fast(throwtarget)
-			visible_message("<span class='danger'>[L] is thrown clear of [src]!</span>")
-	for(var/mob/M in range(7,src))
+			if(L && !qdeleted(L)) // Some mobs are deleted on death
```

Comment:
yep.
Mobs deleting take 2 to 4ds, this is a noticeable lag spike.
At some point, mob will be made to garbage collect.